pysteps/utils/interpolate.py

```diff
@@ -42,186 +41,134 @@ def rbfinterp2d(
-        raise ValueError(
-            "the number of samples in the input_array does not match the "
-            + "number of coordinates %i!=%i" % (npoints, coord.shape[0])
-        )
-
-    # normalize coordinates
-    qcoord = np.percentile(coord, [2, 98], axis=0)
-    dextent = np.max(np.diff(qcoord, axis=0))
-    coord = (coord - qcoord[0, :]) / dextent
-
-    rbfunction = rbfunction.lower()
-    if rbfunction not in _rbfunctions:
-        raise ValueError(
-            "Unknown rbfunction '{}'\n".format(rbfunction)
-            + "The available rbfunctions are: "
-            + str(_rbfunctions)
-        ) from None
-
     # generate the target grid
-    X, Y = np.meshgrid(xgrid, ygrid)
-    grid = np.column_stack((X.ravel(), Y.ravel()))
-    # normalize the grid coordinates
-    grid = (grid - qcoord[0, :]) / dextent
+    xgridv, ygridv = np.meshgrid(xgrid, ygrid)
+    gridv = np.column_stack((xgridv.ravel(), ygridv.ravel()))
+
+    # grid mean resolution
+    x_res = np.gradient(xgrid)
+    y_res = np.gradient(ygrid)
+    mean_res = np.mean(np.abs([x_res.mean(), y_res.mean()]))
 
-    # k-nearest interpolation
-    if k is not None and k > 0:
+    if k is not None:
         k = int(np.min((k, npoints)))
+        tree = _cKDTree_cached(coord, hkey=kwargs.get("hkey", None))
+        dist, inds = tree.query(gridv, k=k)
+        if dist.ndim == 1:
+            dist = dist[..., None]
+            inds = inds[..., None]
+    else:
+        # use all points
+        dist = cdist(coord, gridv, "euclidean").transpose()
+        inds = np.arange(npoints)[None, :] * np.ones((gridv.shape[0], npoints)).astype(
+            int
+        )
 
-        # create cKDTree object to represent source grid
-        tree = scipy.spatial.cKDTree(coord)
+    # compute distance-based weights
+    dist = dist / mean_res + 0.5
+    weights = 1 / np.power(dist, power)
+    weights = weights / np.sum(weights, axis=1, keepdims=True)
 
-    else:
-        k = 0
+    # interpolate
+    output_array = np.sum(
+        input_array[inds, :] * weights[..., None],
+        axis=1,
+    )
 
-    # split grid points in n chunks
-    if nchunks > 1:
-        subgrids = np.array_split(grid, nchunks, 0)
-        subgrids = [x for x in subgrids if x.size > 0]
+    # reshape to final grid size
+    output_array = output_array.reshape(ygrid.size, xgrid.size, nvar)
 
-    else:
-        subgrids = [grid]
+    return np.moveaxis(output_array, -1, 0).squeeze()
 
-    # loop subgrids
-    i0 = 0
-    output_array = np.zeros((grid.shape[0], nvar))
-    for i, subgrid in enumerate(subgrids):
-        idelta = subgrid.shape[0]
 
-        if k == 0:
-            # use all points
-            d = scipy.spatial.distance.cdist(coord, subgrid, "euclidean").transpose()
-            inds = np.arange(npoints)[None, :] * np.ones(
-                (subgrid.shape[0], npoints)
-            ).astype(int)
+@prepare_interpolator()
+def rbfinterp2d(coord, input_array, xgrid, ygrid, **kwargs):

```

Comment:
Same suggestion as above for the variable names

Reply:
done in pysteps/utils/interpolate.py